libs/base/System.idr

```diff
@@ -273,4 +273,12 @@ exitSuccess = exitWith ExitSuccess
 ||| Print the error message and call exitFailure
 export
 die : HasIO io => String -> io a
-die str = do putStrLn str; exitFailure
+die str
+  = do Right () <- fPutStrLn stderr str
+         | Left err => assert_total $ die
```

Comment:
This seems at least half-likely to loop forever given the unlikely event that it fails to print to stderr because I don't know how that could happen and it might be a problem that isn't intermittent.

Reply:
As far as I know, the most likely reason for an error when printing to the `stderr` is the broken pipe error, which won't be cured any time, thus this code will always loop forever. I think, in this case it's better to lose an error message than to loop forever.